File: app/src/main/java/com/github/mbarrben/moviedb/ActivityModule.java
package com.github.mbarrben.moviedb;

import android.app.Activity;
import android.content.Context;
import com.github.mbarrben.moviedb.di.ActivityContext;
import dagger.Module;
import dagger.Provides;

@Module(library = true)
public class ActivityModule {

  private final Activity activity;

  public ActivityModule(Activity activity) {
    this.activity = activity;
  }

  @ActivityContext @Provides Context provideActivityContext() {
    return activity;
  }
}

File: app/src/main/kotlin/com/github/mbarrben/moviedb/di/ActivityModule.kt
package com.github.mbarrben.moviedb.di

import android.app.Activity
import com.github.mbarrben.moviedb.domain.navigation.Navigator
import com.github.mbarrben.moviedb.navigation.NavigatorImpl
import dagger.Module
import dagger.Provides

@Module
class ActivityModule(val activity: Activity) {
  @Provides @PerActivity fun activity(): Activity = activity
  @Provides fun navigator(navigatorImpl: NavigatorImpl): Navigator = navigatorImpl
}